pkg/utils/k8sutil.go

```diff
@@ -453,21 +453,33 @@ func CreateK8sCustomResource(runtime, handler, file, funcName, funcType, topic,
 		Do().Into(&f)
 	if err != nil {
 		if k8sErrors.IsNotFound(err) {
+			funcLabels := map[string]string{}
+			for _, label := range labels {
+				k, v := getKV(label)
```

Comment:
Do the parsing on the calling side (ie: where you're dealing with command line args), that way you won't have to awkwardly re-encode into this form from all the other places you want to call this function (like from unittests).  This function should just accept a `map[string]string` directly, since that's the most natural code-level representation for labels.